tests/test_optimized_pdf_performance.py

```
@@ -0,0 +1,129 @@
+            first_page = min(document.page_mapping.keys())
+            first_page_text = document.get_text_by_page(first_page)
+            if first_page_text and len(first_page_text) > 100:
+                logger.info(f"📖 First Page Sample (100 chars):")
+                logger.info("-" * 50)
+                logger.info(first_page_text[:100] + "...")
+                logger.info("")
+
+        logger.info("✅ Optimized PDF loader test completed successfully!")
+        logger.info(f"💡 Recommendation: {'Streaming was used' if file_info['will_use_streaming'] else 'Standard processing was sufficient'}")
+
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)